scripts/create-postman-collection.ts

```diff
@@ -0,0 +1,177 @@
+            } else {
+                console.log(`no file found for ${resourceName} - ${exampleFilePath}`);
+            }
+        }
+    }
+
+    if (_.has(item, 'item')) {
+        item.item.forEach((child: any) => {
+            updateRequests(child, auth, resourceName, examplesDir);
+        });
+    }
+};
+
+(async () => {
+    try {
+        console.log("let's create this thang.");
+
+        let publicCollectionPath = '';
+        let outputCollectionPath = '';
+        let examplesDir = '';
+        if (process.argv.length < 5) {
+            console.log('please pass in the file paths to your public FHIR collection, examples dir and output file');
+            process.exit(1);
+        }
+        [, , publicCollectionPath, examplesDir, outputCollectionPath] = process.argv;
+
+        // parse the public collection
+        console.log(`parsing public FHIR postman collection, ${publicCollectionPath}`);
+        const publicFHIRContents = await fs.promises.readFile(publicCollectionPath, 'utf-8');
+        const publicCollection = JSON.parse(publicFHIRContents);
+        console.log('parsing complete');
+
+        // load our collection
+        console.log('parsing fwoa postman collection');
+        const fwoaCollectionContents = await fs.promises.readFile(
+            `${__dirname}/../postman/Fhir.postman_collection.json`,
+            'utf-8',
+        );
+        const fwoaCollection = JSON.parse(fwoaCollectionContents);
+        console.log('parsing complete');
+
+        if (!_.has(fwoaCollection, 'item') || fwoaCollection.item.length === 0) {
+            console.log('fwoa postman collection does not have any items defined');
+            process.exit(1);
+        }
+
+        // pluck out the auth from an existing route
+        const fwoaAuthItem = _.find(fwoaCollection.item, item => {
+            return _.has(item, 'request') && _.has(item.request, 'auth') && item.request.auth.type !== 'noauth';
+        });
+        console.log('found fwoa auth');
+        if (_.isUndefined(fwoaAuthItem)) {
+            console.log('fwoa postman collection does not have any requests defined');
+            process.exit(1);
+        }
+        const { auth } = fwoaAuthItem.request;
+
+        // alright, alright, alright let's add any missing items to fwoa collection
+        publicCollection.item.forEach((item: any) => {
+            const resourceName = item.name
+                .split(' ')
+                .join('')
+                .toLowerCase();
+
+            const fwoaItem = _.find(fwoaCollection.item, i => {
+                return i.name === item.name;
+            });
+            if (_.isUndefined(fwoaItem)) {
+                // new item just add after updates
+                const newItem = _.cloneDeep(item);
+
+                updateRequests(newItem, auth, resourceName, examplesDir);
+
+                fwoaCollection.item.push(newItem);
+            } else {
+                // TODO: go recursive here
```

Comment:
Just to understand - this would be how do you handle the situations when FWoA and your public source has a resource? perhaps update the comment to be clearer?

Reply:
Yeah, this would be if both the public collection and fwoa collection have the same top level directory. I didn't want to overwrite the existing requests so there was a thought that we could add any public ones to the same directory. Looks like from upper feedback it's better to just use public + fwoa custom requests so for now they would be one in the same. I can update so if we want to add custom requests to a top level resource we merge public w/fwoa.

Reply:
better comments left in the TODO here with hints on how to refactor and walk both the fwoa collection and public collection trees to do the merge.